spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -181,7 +187,21 @@ class DeltaCatalog extends DelegatingCatalogExtension
       operation.mode,
       writer,
       operation,
-      tableByPath = isByPath).run(spark)
+      tableByPath = isByPath,
+      // We should invoke the Spark catalog plugin API to create the table, to
+      // respect third party catalogs. Note: only handle CREATE TABLE for now, we
+      // should support CTAS later.
+      // TODO: Spark `V2SessionCatalog` mistakenly treat tables with location as EXTERNAL table.
+      //       Before this bug is fixed, we should only call the catalog plugin API to create tables
+      //       if UC is enabled to replace `V2SessionCatalog`.
+      createTableFunc = if (isUnityCatalog && sourceQuery.isEmpty) {
```

Comment:
Is there a way to add unit test for this change? I guess no as UC side changes needs to be released first?

Reply:
Yea, it's a UC-specific hack and we do have tests at the UC side.